### PR TITLE
Move the apply_tensor_props into its own function for more public use

### PR DESCRIPTION
**Pull request by Gamrix**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #67786

Differential Revision: [D32175962](https://our.internmc.facebook.com/intern/diff/D32175962)

### CODE REVIEW

**Comment by Gamrix** [comment]
@gamrix has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32175962).

**Comment by Gamrix** [comment]
@gamrix has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32175962).

File: torch/jit/_static_analysis.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+"""
+
+from typing import Any, List
+
+import torch
+from torch import TensorType
+from torch._C import Graph
+
+
+def apply_input_props_using_example(graph: Graph, example_input: List[Any]):
```
Comment (Krovatkin) [inline torch/jit/_static_analysis.py:15]: This helper only sets dtype and sizes. There are other properties such as strides, device, undefined or not, requires_grad or not. I would rename this or split into two smaller helpers `apply_sizes`, `apply_dtypes`.

**Review by Krovatkin — COMMENTED** [review]
I would consult w/ @gmagogsfm and @eellison if there might be a more appropriate file for this helper(s) .

File: torch/jit/_static_analysis.py (hunk 1 above)
Comment (Gamrix) [reply torch/jit/_static_analysis.py]: How do I get a `TensorType` object from a tensor in Python?

**Comment by Gamrix** [comment]
I messaged @gmagogsfm and updated the location of this file based on his feedback.

**Review by Krovatkin — APPROVED** [review]
:shipit:

**Comment by Gamrix** [comment]
@gamrix has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32175962).

**Pull request closed without merging**